app/src/main/java/com/github/se/orator/model/chatGPT/ChatViewModel.kt

```diff
@@ -228,11 +213,35 @@ class ChatViewModel(
+
+                - **Overall Assessment**: Would you recommend hiring me for the position? Please state explicitly 'Yes, I would recommend hiring you' or 'No, I would not recommend hiring you', and explain your reasoning.
+                - **Strengths**: Highlight the areas where I performed well.
+                - **Weaknesses**: Point out the areas where I need improvement.
+                - **Suggestions for Improvement**: Provide specific advice on how I can enhance my performance in future interviews, especially regarding ${practiceContextAsValue.focusArea}.
+            """
+                    .trimIndent()
             is PublicSpeakingContext ->
-                "Based on my performance, would I win the competition? And say specifically 'Yes I would win the competition' or 'No I would not win the competition'. Please provide reasons for your decision."
+                """
+                Considering my speech prepared for the ${practiceContextAsValue.occasion} with the purpose to ${practiceContextAsValue.purpose}, please provide detailed feedback including:
+
+                - **Overall Assessment**: How effective was my delivery? Did I successfully achieve my speech's purpose? Please state explicitly 'Yes, you were effective' or 'No, you were not effective', and explain your reasoning.
+                - **Strengths**: Highlight the aspects of my speech that were particularly strong.
+                - **Weaknesses**: Identify areas where I could improve.
+                - **Suggestions for Improvement**: Provide specific advice on how I can enhance my speech, focusing on ${practiceContextAsValue.focusArea} and ${practiceContextAsValue.feedbackType}.
+            """
+                    .trimIndent()
             is SalesPitchContext ->
-                "Based on my performance, did I successfully convince you in the negotiation? And say specifically 'Yes you successfully convinced me' or 'No you did not convince me'. Please provide reasons for your decision."
+                """
+                Based on my sales pitch for "${practiceContextAsValue.product}" aimed at ${practiceContextAsValue.targetAudience}, please provide detailed feedback including:
+
+                - **Overall Assessment**: Did I achieve my sales goal to ${practiceContextAsValue.salesGoal}? Please state explicitly 'Yes, you achieved your sales goal' or 'No, you did not achieve your sales goal', and explain your reasoning.
+                - **Strengths**: Highlight the areas where my pitch was effective.
+                - **Weaknesses**: Point out aspects where I could improve.
+                - **Suggestions for Improvement**: Provide specific advice on how I can enhance my sales pitch, especially regarding ${practiceContextAsValue.negotiationFocus} and ${practiceContextAsValue.feedbackType}.
+            """
+                    .trimIndent()
```

Comment:
These updates make a significant improvement in providing detailed contextual guidelines and tailored feedback logic for each practice scenario.

---

#### **Contextual Guidelines**
- The guidelines for `InterviewContext`, `PublicSpeakingContext`, and `SalesPitchContext` are detailed and highly relevant, ensuring a professional and realistic simulation for the user.
- The use of placeholders like `$practiceContextAsValue.*` effectively integrates dynamic information, making the content adaptable to specific scenarios.
- The division into sections like "Question Structure," "Focus Areas," and "Post-Session Feedback" provides a clear and logical structure that makes it easy for users to follow and engage with the practice.

---

#### **Feedback Logic**
- Feedback logic categorizes responses into strengths and areas for improvement, offering a balanced approach that motivates users while providing actionable suggestions.
- The feedback is tailored to the specific context (`Interview`, `PublicSpeaking`, or `SalesPitch`), ensuring relevance and engagement.
- The inclusion of questions like "Would you recommend hiring me for the position?" and "Did I achieve my sales goal?" encourages reflection and aligns the feedback with realistic performance evaluation criteria.

---

### **Overall Observations**
These updates significantly enhance the functionality and user experience by making the sessions feel more structured, professional, and personalized. Excellent work!

Reply:
Thanks!